src/client/public/scripts/models/client-id.ts

```diff
@@ -0,0 +1,73 @@
+const DB_NAME = 'project-health-analytics';
+const DB_VERSION = 1;
+const CLIENT_ID_STORENAME = 'client-id-store';
+const CLIENT_ID_KEY = 'clientID';
+
+export class ClientIDModel {
+  private db: IDBDatabase|null;
+
+  constructor() {
+    this.db = null;
+  }
+
+  private async getDB(): Promise<IDBDatabase> {
+    if (this.db) {
+      return this.db;
+    }
+
+    return new Promise((resolve, reject) => {
+             const request = indexedDB.open(DB_NAME, DB_VERSION);
+             request.onerror = () => {
+               reject(request.error);
+             };
+             request.onupgradeneeded = this.updateDB;
+             request.onsuccess = (event) => {
+               // tslint:disable-next-line:no-any
+               this.db = (event.target as any).result as IDBDatabase;
+               resolve(this.db);
+             };
+           }) as Promise<IDBDatabase>;
+  }
+
+  // tslint:disable-next-line:no-any
+  private async updateDB(event: any) {
+    const db = event.target.result;
+    if (event.oldVersion < 2) {
```

Comment:
What does this mean?

Reply:
If the oldVersion was < 2 (i.e. it's new), perform these steps and with future versions will have `< XYZ` and perform further upgrades.

Basically this sets up support for future changes.

Reply:
I see.